bucket/jd-gui.json

```diff
@@ -1,15 +1,26 @@
 {
-    "homepage": "http://jd.benow.ca/",
+    "homepage": "https://github.com/java-decompiler/jd-gui",
+    "description": "A standalone Java Decompiler GUI.",
+    "version": "1.4.2",
     "license": "GPL-3.0-only",
-    "version": "1.4.0",
-    "url": "https://github.com/java-decompiler/jd-gui/releases/download/v1.4.0/jd-gui-windows-1.4.0.zip",
-    "hash": "457713a4e08b1c7333a09efa4317f873d7e530fd19ccdb66e585d553a263ee26",
-    "bin": "jd-gui.exe",
-    "extract_dir": "jd-gui-windows-1.4.0",
+    "hash": "c2f11ee24e35fb2ca6ebbc1f002ec447095f140da9b58fac46a388c69867aaa2",
+    "url": "https://github.com/java-decompiler/jd-gui/releases/download/v1.4.2/jd-gui-1.4.2.jar#/jd-gui.jar",
```

Comment:
Why did you change it to the `.jar` file?

Reply:
jd-gui uses [gradle-launch4j](https://github.com/TheBoegl/gradle-launch4j) to config launch4j so that jd-gui can distribute a standalone exe file(See [build.gradle](https://github.com/java-decompiler/jd-gui/blob/master/build.gradle#L7) of jd-gui). It forces jd-gui to running on JRE8 **only**. 

For me, it's really unnecessary - I have to install a JRE8 runtime for it. And jd-gui works fine on JRE 9+. Besides, JRE8 didn't support HiDPI.

Reply:
Sounds reasonable. :smile:
The following should also work:
```json
    "bin": "jd-gui.jar",
    "shortcuts": [
        [
            "jd-gui.jar",
            "Java Decompiler"
        ]
    ]
```

Reply:
No. Jar files as shortcut do not work

Reply:
When you open it from start menu you are asked for select default handler for .jar files

Reply:
It works for cmd/powershell, but not from start menu, if JRE/JDK is installed from scoop(like me), because the `.jar` file is not associated with java runtime.

Reply:
After selecting `javaw.exe` it will work. But the `.bat` uses the current sourced Java version which makes more sense.

Reply:
Shortcut creation could be extended for these jar files (There are lots of them in repositories) to automatically "shim" (just automatically create bat with argument handling and somehing like (scoop which javaw, or directly call file_path function)) jar files first into installation directory and then create shortcut from this shim.